localstack-core/localstack/services/s3/utils.py

```diff
@@ -390,40 +412,6 @@ def get_full_default_bucket_location(bucket_name: BucketName) -> str:
         return f"{config.get_protocol()}://{bucket_name}.s3.{host_definition.host_and_port()}/"
 
 
-def get_object_checksum_for_algorithm(checksum_algorithm: str, data: bytes) -> str:
-    match checksum_algorithm:
-        case ChecksumAlgorithm.CRC32:
-            return checksum_crc32(data)
-
-        case ChecksumAlgorithm.CRC32C:
-            return checksum_crc32c(data)
-
-        case ChecksumAlgorithm.SHA1:
-            return hash_sha1(data)
-
-        case ChecksumAlgorithm.SHA256:
-            return hash_sha256(data)
-
-        case _:
-            # TODO: check proper error? for now validated client side, need to check server response
-            raise InvalidRequest("The value specified in the x-amz-trailer header is not supported")
-
-
-def verify_checksum(checksum_algorithm: str, data: bytes, request: Dict):
-    # TODO: you don't have to specify the checksum algorithm
-    # you can use only the checksum-{algorithm-type} header
-    # https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html
-    key = f"Checksum{checksum_algorithm.upper()}"
-    # TODO: is there a message if the header is missing?
-    checksum = request.get(key)
-    calculated_checksum = get_object_checksum_for_algorithm(checksum_algorithm, data)
-
-    if calculated_checksum != checksum:
-        raise InvalidRequest(
-            f"Value for x-amz-checksum-{checksum_algorithm.lower()} header is invalid."
-        )
-
-
```

Comment:
this code wasn't used anymore anywhere, since we're properly calculating checksums on streams instead of the full string